### ISSUE TITLE
status-state

### ISSUE THREAD

**Issue by Frankhendriks** [ICEPAY/Magento-2#8]
Is it possible to set/asign a status-state to payments done with icepay? i need them to be the same state as the default ''bank transfer'' status-state

**Comment by tsygankov** [comment]
Hi Frankhendriks,

You can configure order statuses in **Admin Panel ->Stores > Settings > Order Status**?

You can find more information [here](https://www.mageplaza.com/kb/magento-2-order-status-order-state.html)

ICEPAY order status codes are:
icepay_icpcore_new 
icepay_icpcore_open 
icepay_icpcore_ok
icepay_icpcore_error

Please let me know if that answers your question.

/Stan

**Comment by Frankhendriks** [comment]
but in Admin Panel ->Stores > Settings > Order Status i can only set order status for standard magento payment methods and not for icepay payments.
when i do a normal magento order with payment option ( bank transfer ) they get the status "new'' but when i do an icepay payment method they go to processing. but it should also be new.
on a side note, i updated the plugin to latest version but i do not have those icepay status codes.

**Comment by Frankhendriks** [comment]
so strange, for some reason i am missing the 4 icepay statusses.
I will now try to remove the module and install it again, i'll also clear the database tables.

**Comment by tsygankov** [comment]
You can create ICEPAY status codes manually or you can try running the [UpgradeData ](https://github.com/ICEPAY/Magento-2/blob/master/Setup/UpgradeData.php)script.

You can temporary change 
`if (version_compare($context->getVersion(), '1.0.2') < 0)`
to
` if(true) `

to force the data upgrade.

**Comment by Frankhendriks** [comment]
running the UpgradeData.php script with your adjustment gives me this error:
PHP Fatal error:  Interface 'Magento\Framework\Setup\InstallSchemaInterface' not found in /data/www/gastrodeals.nl/public_html/app/code/Icepay/IcpCore/Setup/UpgradeData.php on line 18

**Comment by tsygankov** [comment]
Sorry, I forgot to mention, you need to run 
`php bin/magento setup:upgrade`

You can also try changing values in the **setup_module** table. I haven't tested, but probably it will work:

1) UPDATE setup_module SET  data_version = '1.0.1' where module = 'Icepay_IcpCore';
2) run the `php bin/magento setup:upgrade` command.

**Comment by Frankhendriks** [comment]
yea the thing is, when i was in 1.0.1 and i wanted to upgrade to 1.0.2 it gave the version error so i had to manually change the value in setup_module to get the extention to work.

**Comment by Frankhendriks** [comment]
there is not really an error message, it simply does not update the data_version to 1.0.2
so it says in frontend something like this:
The following modules are outdated:
Icepay_IcpCore schema: current version - 1.0.1, required version - 1.0.2
i changed the number in the database and it worked.

**Comment by tsygankov** [comment]
Have you tried to run `php bin/magento setup:upgrade` ? What was the output? 

**setup:upgrade** command should upgrade both schema and data. If it doesn't work then probably there is a bug somewhere.

**Comment by Frankhendriks** [comment]
the setup:upgrade script just runs its course, no errors whatsoever, except it does not update the data_version. it's super strange.

**Comment by tsygankov** [comment]
You can try changing module version to 1.0.3 in etc/module.xml
`<module name="Icepay_IcpCore" setup_version="1.0.3">`

and adjusting UpgradeSchema.php and and UpgradeData.php:

`if (version_compare($context->getVersion(), '1.0.3') < 0) {`

It is possible that UpgradeSchema.php fails (can't drop the foreign key) revert the changes in this file and run the `php bin/magento setup:upgrade` command again

In the next release setup_version will change to 1.1.x, so changing setup_version to 1.0.3 won't brake compatibility with newer module versions.

**Comment by Frankhendriks** [comment]
so i seem to finally have it working, order states and statusses were added.
bank transfer and a online payment are both state ''new'' but different status but that is no problem.
now 1 thing that is strange, the bank transfers send an automatic email order confirmation, but the online payed ones do not send the order confirmation email.

**Comment by Frankhendriks** [comment]
Did you change postback url between versions ? 
i had to change : 
http://acc.gastrodeals.nl/icepay/postback/notification
into:
http://acc.gastrodeals.nl/rest/V1/icepay/postback
but now transaction details are not being posted.

**Comment by tsygankov** [comment]
Yes, the postback handler was [moved ](https://github.com/ICEPAY/Magento-2/commit/c13ea4b69f35af447411523736edf5c2b3be910f )to REST API.

> now 1 thing that is strange, the bank transfers send an automatic email order confirmation, but the online payed ones do not send the order confirmation email.

I will check it out and get back to you.

**Comment by NataljaCig** [comment]
Hi Frankhendriks, 
We are planning to add the transaction details to the next module version. We are working hard to push changes to the repository by the end of the week.

**Comment by Frankhendriks** [comment]
Hey natalja, 

That is good to hear, is there any news ? did you update the repository ?

**Comment by NataljaCig** [comment]
Hi Frankhendriks,
Yes. See the Dev brunch. The lead developer reported that he introduced many minor changes and some bugfixes and merged to Dev branch version 1.2.0 yesterday. The problem with missing transactions should be solved by now; however, we plan to continue testing, reworking and improving this part of the module in future. At the moment, the testing has not been completely finished. It would be really good if you could independently test and confirm if the problem is solved.

**Comment by Frankhendriks** [comment]
aaaah, so what would be your advice, we are already using our site in live environment right now. payments are coming in correctly but i do manual sending the order confirmation email etc.

**Comment by NataljaCig** [comment]
Please let me know which version you're using now (master or Dev branch). I would recommend to wait for few more days until we finish regression testing though we can't guarantee that we will release the stable version immediately after testing because we don't know yet how many bugs we could find. If the problem is really urgent the developer can create for you a separate branch with adjustments in order status update logic.

**Comment by Frankhendriks** [comment]
only 1.0.1 was working correctly for me.
i might still need some adjustments to order status update logic, that a new order that has been payed stays in state: new status : processing so our ERP system can read in the new orders and then put them in state: processing.

**Comment by Frankhendriks** [comment]
Any update on when code will be pushed to master?

**Comment by tsygankov** [comment]
Hi Frankhendriks,

Can you please create a zip archive with the module code that you have in production and send it to me by the [email](mailto:modules@icepay.com) (or you can specify the hash of the commit)? Module version that we have in the master branch uses Magento REST API for postback notifications, but I guess you need an adjustment for an older version of the module.
